daisy/resource_registry.go

```diff
@@ -236,3 +238,30 @@ func (r *baseResourceRegistry) regUse(name string, s *Step) (*Resource, DError)
 	r.m[name].users = append(r.m[name].users, s)
 	return res, nil
 }
+
+// regUseDeviceName registers a Step s as a user of a disk device resource.
+// "DeviceName" is only used by DetachDisks API.
+func (r *baseResourceRegistry) regUseDeviceName(name string, s *Step) (*Resource, DError) {
```

Comment:
What is the need of this func compared to regUse? 

Wouldn't the regUse line:
if r.urlRgx != nil && r.urlRgx.MatchString(name) {

Just be ignored when dd.DeviceName is sent and the two functions would work the same, except for the "return nil, nil" line?